mava/core_jax.py

```diff
@@ -69,3 +69,59 @@ def launch(
                 are primarily for debugging
             name : name of the system
         """
+
+
+class SystemBuilder(abc.ABC):
+    """Abstract system builder."""
+
+    @abc.abstractmethod
+    def data_server(self) -> List[Any]:
+        """Data server to store and serve transition data from and to system.
+
+        Returns:
+            System data server
+        """
+
+    @abc.abstractmethod
+    def parameter_server(self) -> Any:
```

Comment:
Are these returning properties? E.g. `return self._parameter_server` ? If so we should add `@property` to the definition.